daft/functions/process.py

```diff
@@ -0,0 +1,98 @@
+"""Process Functions."""
+
+from __future__ import annotations
+
+import subprocess
+from typing import Any, Literal
+
+import daft
+from daft.datatype import DataType, DataTypeLike
+from daft.expressions import Expression
+
+
+def _convert_stdout(stdout: str | None, return_dtype: DataTypeLike) -> Any:
+    """Convert stdout string to Python value matching return_dtype when possible."""
+    if stdout is None:
+        return None
+
+    inferred_dtype = DataType._infer(return_dtype)
+    if inferred_dtype == DataType.string():
+        return stdout
+    if inferred_dtype == DataType.int64() or inferred_dtype == DataType.int32():
+        return int(stdout)
+    if inferred_dtype == DataType.float64() or inferred_dtype == DataType.float32():
+        return float(stdout)
+    if inferred_dtype == DataType.bool():
+        normalized_stdout = stdout.strip().lower()
+        if normalized_stdout in ("true", "1", "t", "yes"):
+            return True
+        if normalized_stdout in ("false", "0", "f", "no"):
+            return False
+        return None
+
+    return stdout
+
+
+def run_process(
+    args: Expression | list[Expression | Any],
+    *,
+    shell: bool = False,
+    on_error: Literal["raise", "ignore", "log"] = "log",
+    return_dtype: DataTypeLike = DataType.string(),
+) -> Expression:
+    """Returns an expression that runs an external process (optionally via a shell) and exposes its stdout as a column.
+
+    This helper wraps a Python UDF around ``subprocess.run`` so it can be used inside DataFrame expressions.
+
+    Args:
+        args (Expression | list[Expression | Any]):
+            The command to execute. If ``shell=False`` (default), pass a list of arguments, for example ``["ls", "-a", col("path")]``.
+            If ``shell=True``, pass a single string expression, for example ``format("echo {}", col("x"))``.
```

Comment:
We could technically support a single string argument for `shell=False` and multiple arguments for `shell=True`. Maybe we can just follow the behavior of Python [subprocess.Popen](https://docs.python.org/3/library/subprocess.html#popen-constructor) here.

Reply:
Thanks . I change it.